### DISCUSSION TITLE
How to rewrite static files automatically?

### DISCUSSION THREAD

**Discussion by GodderE2D** [vercel/vercel#6242] [Help] [answered]
I have static .html files and would like to remove the file extension when visiting the website. I found out you can do this using rewrites, however, I have to do each file manually, and when you have tons of pages, it's kind of tiring. Is there some sort of regex or wildcard I can use? Or anything else that can solve this problem? Thanks.

Here is my current vercel.json file:
`{
  "rewrites": [
    { "source": "/about", "destination": "/about.html" },
    { "source": "/rewrite", "destination": "/rewrite.html" },
    { "source": "/catch", "destination": "/catch.html" },
    { "source": "/branding", "destination": "/branding.html" },
    { "source": "/discord", "destination": "/discord.html" },
    { "source": "/api", "destination": "/api.html" }
    ]
}
`
...and so on. (Sorry if there are no newlines, I don't have any files to supply, and markdown is weird)

Thanks!

**Comment by mauricewegner** [comment, marked as answer]
You might be looking for the `cleanUrls` (see [here](https://vercel.com/docs/configuration#project/clean-urls)) option.

**Reply by GodderE2D** [reply]
Thanks! This works very well.

**Comment by fu-sen** [comment]
I think `cleanUrls` can solve it, but if you need `rewrite`, there are some patterns you can use:
https://github.com/isaacs/node-glob#glob-primer
